content/token22-in-the-client.md

```diff
@@ -0,0 +1,470 @@
+
+When interacting with mints and tokens, we need to be sure we're using the correct Token Program. To create a `Token Program` mint, use `Token Program`; to create a mint with extensions, use the `Token Extension Program`.
+
+Fortunately, the `spl-token` package makes it simple to do this. It provides both the `TOKEN_PROGRAM_ID` and `TOKEN_2022_PROGRAM_ID` constants, along with all of its helper functions for creating and minting tokens that take a program ID as input.
+
+NOTE: `spl-token` defaults to using the `TOKEN_PROGRAM_ID` unless otherwise specified. Make sure to explicitly pass the `TOKEN_2022_PROGRAM_ID` for all function calls related to the Token Extensions Program. Otherwise, you will get the following error: `TokenInvalidAccountOwnerError`.
+
+## Considerations when working with both Token and Extension Tokens
+
+Although the interfaces for both of these programs remain consistent, they are two different programs. The program IDs for these programs are unique and non-interchangeable, resulting in distinct addresses when utilized. If you want to support both `Token Program` tokens and `Token Extension Program` tokens, you must add extra logic on the client side.
+
+## Associated Token Accounts (ATA)
+
+An Associated Token Account (ATA) is a Token Account whose address is derived using the wallet's public key, the token's mint, and the token program. This mechanism provides a deterministic Token Account address for each mint per user. The ATA account is usually the default account for most holders. Fortunately, ATAs are handled the same way with both token programs.
+
+We can use the ATA helper functions for each token program by providing the desired program ID. 
+
+If we want to use the Token Extension Program when we call `getOrCreateAssociatedTokenAccount` Extension Tokens, we can pass in `TOKEN_2022_PROGRAM_ID` for the `tokenProgramId` parameter:
+
+```ts
+const tokenProgramId = TOKEN_2022_PROGRAM_ID;
+
+const tokenAccount = await getOrCreateAssociatedTokenAccount(
+  connection,
+  payer,
+  mintAddress,
+  payer.publicKey,
+  true,
+  'finalized',
+  {commitment: 'finalized'},
+  tokenProgramId // TOKEN_PROGRAM_ID for Token Program tokens and TOKEN_2022_PROGRAM_ID for Token Extension Program tokens
+)
+```
+
+To re-create the ATA's address from scratch, we can use the `findProgramAddressSync` function by providing the correct seeds:
+
+```ts
+function findAssociatedTokenAddress(
+  walletAddress: PublicKey,
+  tokenMintAddress: PublicKey
+): PublicKey {
+  return PublicKey.findProgramAddressSync(
+    [
+      walletAddress.toBuffer(),
+      TOKEN_PROGRAM_ID.toBuffer(), // replace TOKEN_PROGRAM_ID with TOKEN_2022_PROGRAM_ID for Token22 tokens
+      tokenMintAddress.toBuffer(),
+    ],
+    ASSOCIATED_TOKEN_PROGRAM_ID
+  )[0];
+}
+```
+
+## How to fetch tokens
+
+When it comes to fetching tokens, there is no difference between tokens and extension tokens. All we have to do is provide the correct token program:
```

Comment:
Suggest: "there is no difference between tokens made with Token Program or Token Extensions Program." As 'tokens' may mean either.